### PR TITLE
use manual fold function for list of hosts

### PR DESCRIPTION
**Pull request by Pacman99**
mapAttrs'/listToAttrs doesn't merge outputs properly. This creates a function, foldHosts, that combines the hosts into their respective flake outputs properly and doesn't cut hosts.

### CODE REVIEW

File: systemFlake.nix [hunk 1]
```diff
@@ -27,6 +27,7 @@
 
 let
   inherit (flake-utils-plus.lib) eachSystem;
+  inherit (builtins) foldl' mapAttrs removeAttrs attrValues;
```
Comment (gytis-ivaskevicius) [inline systemFlake.nix:30]: Good stuff :ok_hand:

File: systemFlake.nix [hunk 2]
```diff
@@ -66,14 +67,26 @@ let
       extraArgs = sharedExtraArgs // lhs.extraArgs // rhs.extraArgs;
     };
 
+  foldHosts = 
+    let 
+      mergeOutput = lhs: rhs:
```
Comment (gytis-ivaskevicius) [inline systemFlake.nix:72]: I am fine with it - but i doubt that there is any need to move out this `mergeOutput` function to `let` statement

**Review by gytis-ivaskevicius — APPROVED** [review]

**Comment by gytis-ivaskevicius** [comment]
Merging this PR since I miss having a stable staging branch :D Let's keep in mind the "considerations" above for future code changes

File: systemFlake.nix [hunk 3]
```diff
@@ -66,14 +67,26 @@ let
...
+        let output = rhs.name; in
+        lhs // { 
+          # manually merge whats inside the output to prevent // override
+          ${output} = lhs.${output} or { } // rhs.value;
```
[the remark this answers is not in the document]
Comment (Pacman99) [reply systemFlake.nix:76]: This was just for readability. If you read the fold function without reading configurationBuilder, its not really clear what rhs.name is.

File: systemFlake.nix (hunk 2 above)
Comment (Pacman99) [reply systemFlake.nix:72]: good point. I did it originally because of the `getOutput` function, which is no longer there.

**Pull request merged**